### ISSUE TITLE
ChromeOS: git operation on the local repo that's under Google-Drive acts strange

### ISSUE THREAD

**Issue by 130s** [kinu-garage/hut_10sqft#962]
On brya, weird `git` operation result has been seen incl.:
- Squashing multiple commits by `git rebase` only creates a new commit, instead reduce.
- Endless `git rebase --continue` failure without reasonable erroneous situation.

**Comment by 130s** [comment]
I'm now suspecting it's because of conflict that is caused by Google-Drive that syncs with cloud but only with lag? IDK. Sync tool should take precendence of newer modification, so if changes are only made locally then this hypothesis doesn't make sense. However, I'm out of the idea.

Will try relocating the git config repo out of G-Drive path.

**Comment by 130s** [comment]
I confirmed the issue.

For the same local repo, the operation I kept failing (`git rebase -i HEAD~3` then squash) on the path that synchs with G-Drive does not fail on non-synched path.

I don't know why though. Why `git` operation on the local repo fails when files are synching with G-Drive, but NOT fails when synching with Dropbox (on Ubuntu)? Maybe there are something I'm missing on file system and/or others on ChromeOS that are affecting `git`?

**Comment by 130s** [comment]
TLDR; Moved `hut_10sqft` repo out of G-Drive to `~/.config/hut_10sqft` where the repo needs to be cloned into.

Tired of digging. Reviewing the design-4 https://github.com/kinu-garage/hut_10sqft/issues/900#issuecomment-1737972615 and the considerations for other design options in that ticket, I found the benefit in synching git repo on G-Drive is limited, and yet disadvantage is huge as already discussed in this ticket. 

<details><summary>E.g. file permission on a git repo on a local G-Drive</summary>

```
chromeos_host_user@penguin:~/link/GoogleDrive/computer/hut_10sqft/hut_10sqft/config/bash$ ll 
total 83K  
drwxr-x--- 2 chromeos_host_user chronos-access 4.0K Jan 16 16:27 ./
drwxr-x--- 6 chromeos_host_user chronos-access 4.0K Jan 16 16:18 ../                 
-rw-r----- 1 chromeos_host_user chronos-access 1.1K Jan 16 15:59 bash_alias          
-rw-r----- 1 chromeos_host_user chronos-access  107 Jan 16 15:59 bash_application_130s-mac1.bash       
-rw-r----- 1 chromeos_host_user chronos-access  178 Jan 16 15:59 bash_application_130s-p50.bash        
-rw-r----- 1 chromeos_host_user chronos-access   57 Jan 16 15:59 bash_application_btt.bash         
-rw-r----- 1 chromeos_host_user chronos-access  822 Jan 16 15:59 bash_application_btt-osrf.bash        
-rw-r----- 1 chromeos_host_user chronos-access  886 Jan 16 15:59 bash_application_common.bash      
-rw-r----- 1 chromeos_host_user chronos-access  518 Jan 16 15:59 bash_application_elwing.bash          
-rw-r----- 1 chromeos_host_user chronos-access   80 Jan 16 15:59 bash_application_mac1.bash        
-rw-r----- 1 chromeos_host_user chronos-access  178 Jan 16 15:59 bash_application_serval.bash
-rw-r----- 1 chromeos_host_user chronos-access  109 Jan 16 15:59 bashrc_130s-kudu1
-rw-r----- 1 chromeos_host_user chronos-access   96 Jan 16 15:59 bashrc_130s-p50
-rw-r----- 1 chromeos_host_user chronos-access  110 Jan 16 15:59 bashrc_130s-serval
-rw-r----- 1 chromeos_host_user chronos-access  104 Jan 16 15:59 bashrc_130s-t440s
-rw-r----- 1 chromeos_host_user chronos-access 6.4K Jan 16 15:59 bash_setup_common.bash
-rw-r----- 1 chromeos_host_user chronos-access  54K Jan 16 15:59 git-completion.bash
-rw-r----- 1 chromeos_host_user chronos-access  314 Jan 16 15:59 mac_common.bash
-rw-r----- 1 chromeos_host_user chronos-access   97 Jan 16 15:59 rc_130s-kudu1.bash
-rw-r----- 1 chromeos_host_user chronos-access  148 Jan 16 15:59 rc_130s-p50.bash
-rw-r----- 1 chromeos_host_user chronos-access  276 Jan 16 15:59 rc_130s-serval.bash
-rw-r----- 1 chromeos_host_user chronos-access  184 Jan 16 15:59 rc_130s-w510.bash
-rwxr-x--- 1 chromeos_host_user chronos-access   98 Jan 16 15:59 rc_t440s.bash*
-rw-r----- 1 chromeos_host_user chronos-access 1.8K Jan 16 15:59 ubuntu_common.bash
```
</details>

<details><summary>E.g. file permission on a git repo on a non G-Drive path on ChromeOS</summary>

```
gm130s@penguin:/tmp/hut_10sqft$ ll
total 28K
drwxr-x--- 1 gm130s gm130s  254 Jan 16 15:58 .git/       
drwxr-x--- 1 gm130s gm130s  172 Jan 16 15:56 test/       
drwxr-x--- 1 gm130s gm130s  172 Jan 16 15:56 ./          
drwxr-x--- 1 gm130s gm130s   42 Jan 16 15:56 src/        
-rw-r----- 1 gm130s gm130s 1.6K Jan 16 15:56 package.xml 
drwxr-x--- 1 gm130s gm130s  216 Jan 16 15:56 executable/ 
-rw-r----- 1 gm130s gm130s  666 Jan 16 15:56 README.rst  
-rw-r----- 1 gm130s gm130s  12K Jan 16 15:56 LICENSE     
-rw-r----- 1 gm130s gm130s  788 Jan 16 15:56 .travis.yml
-rw-r----- 1 gm130s gm130s  544 Jan 16 15:56 .gitignore
drwxr-x--- 1 gm130s gm130s  754 Jan 16 15:56 config/
drwxr-x--- 1 gm130s gm130s   18 Jan 16 15:56 hut_10sqft/
drwxrwxrwt 1 root   root    446 Jan 16 15:56 ../
```
</details>

New plan should be called Design-5.

### Design-5

```
Linux-ChromeOS
|- /home/$USER/.config
   |- hut_10sqft
   :
```
- o Can avoid `git` to conflict with another version control process i.e. Google Drive's local process, which would turn files uncontrollable  https://github.com/kinu-garage/hut_10sqft/issues/962
- o Can control when to apply the changes made on other hosts, made for other type of OSes.
- x Manual maintenance needed on every ChromeOS-Linux hosts. The changes made on other hosts will no longer automatically be applied as they used to be.
   - If I add more ChromeOS-Linux hosts (i.e. ChromeOS where I enable Linux mode), manual maintenance on each will be required.
